packages/tabs/src/tab-nav.vue

```diff
@@ -0,0 +1,376 @@
+<script lang='ts'>
+import { h, defineComponent, ref, inject, computed, onUpdated, onMounted, onBeforeUnmount } from 'vue'
+import { addResizeListener, removeResizeListener } from '@element-plus/utils/resize-event'
+import { firstUpperCase } from '@element-plus/utils/util'
+import TabBar from './tab-bar.vue'
+import { NOOP } from '@vue/shared'
+
+export default defineComponent({
+  name: 'ElTabNav',
+  components: {
+    TabBar,
+  },
+  props: {
+    panes: {
+      type: Array,
+      default: () => ([]),
+    },
+    currentName: {
+      type: String,
+      default: '',
+    },
+    editable: Boolean,
+    onTabClick: {
+      type: Function,
+      default: NOOP,
```

Comment:
Common mistake here, default should've been
```ts
default: () => Noop
```
if you choose to use this like
```ts
default: NOOP
```
and the user did not specify `onTabClick` prop, then when you try to invoke `onTabClick` it will throw error because `onTabClick` is resolved as value `undefined` or `void` because `NOOP` were evaluated and returned nothing